lib/mayaUsd/python/wrapPrimWriter.cpp

```diff
@@ -246,10 +262,10 @@ class PrimWriterWrapper
 
     static void Register(boost::python::object cl, const std::string& mayaTypeName)
     {
-        UsdMayaPrimWriterRegistry::WriterFactoryFn fn
-            = FactoryFnWrapper::Register(cl, mayaTypeName);
-        if (fn) {
-            UsdMayaPrimWriterRegistry::Register(mayaTypeName, fn, true);
+        bool             updated = false;
+        FactoryFnWrapper fn = FactoryFnWrapper::Register(cl, mayaTypeName, updated);
+        if (!updated) {
+            UsdMayaPrimWriterRegistry::Register(mayaTypeName, fn, fn, true);
         }
```

Comment:
I guess the fact the class has an operator() allows it to be both the predicate and factory function, but a comment explaining why fn gets passed twice would make the code clearer